.github/workflows/pytest-gpu.yml

```diff
@@ -49,68 +49,46 @@ jobs:
-
-    - name: Install dependencies
-      run: |
-        pip3 install --upgrade pip
-        pip3 install -e .[extras]
+          docker build . --file docker/Dockerfile.devito --tag devito_img --build-arg base=${{ matrix.base }}
 
     - name: Test with pytest
       run: |
-        ${{ matrix.test_drive_cmd }}
-        pytest --cov --cov-config=.coveragerc --cov-report=xml tests/test_adjoint.py tests/test_gpu_common.py
-        pytest --cov --cov-config=.coveragerc --cov-report=xml ${{ matrix.test_file }}
+        docker run --gpus all --rm --name testrun devito_img pytest --cov --cov-config=.coveragerc --cov-report=xml tests/test_adjoint.py tests/test_gpu_common.py ${{ matrix.test_file }}
 
     - name: Test examples
       run: |
-        pytest examples/seismic/acoustic/acoustic_example.py
-        pytest examples/seismic/elastic/elastic_example.py
-        pytest examples/seismic/tti/tti_example.py
-        pytest examples/seismic/viscoacoustic/viscoacoustic_example.py
-        pytest examples/seismic/viscoelastic/viscoelastic_example.py
+        docker run --gpus all --rm --name testrun devito_img pytest examples/seismic/acoustic/acoustic_example.py
+        docker run --gpus all --rm --name testrun devito_img pytest examples/seismic/elastic/elastic_example.py
+        docker run --gpus all --rm --name testrun devito_img pytest examples/seismic/tti/tti_example.py
+        docker run --gpus all --rm --name testrun devito_img pytest examples/seismic/viscoacoustic/viscoacoustic_example.py
+        docker run --gpus all --rm --name testrun devito_img pytest examples/seismic/viscoelastic/viscoelastic_example.py
 
     - name: Test examples with MPI
-      if: matrix.name != 'pytest-gpu-omp-amd'
       run: |
-        DEVITO_MPI=1 mpirun -n 2 pytest examples/seismic/acoustic/acoustic_example.py
-        DEVITO_MPI=1 mpirun -n 2 pytest examples/seismic/elastic/elastic_example.py
-        DEVITO_MPI=1 mpirun -n 2 pytest examples/seismic/tti/tti_example.py
-        DEVITO_MPI=1 mpirun -n 2 pytest examples/seismic/viscoacoustic/viscoacoustic_example.py
-        DEVITO_MPI=1 mpirun -n 2 pytest examples/seismic/viscoelastic/viscoelastic_example.py
+        docker run --gpus all --rm --name testrun --env DEVITO_MPI=1 devito_img mpirun -n 2 pytest examples/seismic/acoustic/acoustic_example.py
+        docker run --gpus all --rm --name testrun --env DEVITO_MPI=1 devito_img mpirun -n 2 pytest examples/seismic/elastic/elastic_example.py
+        docker run --gpus all --rm --name testrun --env DEVITO_MPI=1 devito_img mpirun -n 2 pytest examples/seismic/tti/tti_example.py
+        docker run --gpus all --rm --name testrun --env DEVITO_MPI=1 devito_img mpirun -n 2 pytest examples/seismic/viscoacoustic/viscoacoustic_example.py
+        docker run --gpus all --rm --name testrun --env DEVITO_MPI=1 devito_img mpirun -n 2 pytest examples/seismic/viscoelastic/viscoelastic_example.py
 
     - name: Upload coverage to Codecov
```

Comment:
Perhaps a silly question...

I wonder whether / how will codecov work? 

The code coverage collection now runs in a docker container, how will it see it?

Reply:
Not sure we'll see when it runs